test/libraries/CalldataDecoder.t.sol

```diff
@@ -83,6 +87,56 @@ contract CalldataDecoderTest is Test {
         _assertEq(_config, config);
     }
 
+    function test_fuzz_decodeSwapExactInParams(IV4Router.ExactInputParams calldata _swapParams) public view {
```

Comment:
Ok hm I just had this thought which I think relates to all of these tests... Basically we encode the fuzz, and decode it with the library. But we never decode it regularly... shouldnt we be testing it against abi.decode9) not just the raw passed in data? 

ie 

_encodedParams = abi.encode(_swapParams)
_params = abi.decode(_encodedParams, IV4Router.ExactInputParams)

and then all the asserts are against _params

but lmk if you dont agree

Reply:
#251